packages/wrangler/src/config/validation.ts

```diff
@@ -3221,6 +3230,44 @@ const validateMigrations: ValidatorFn = (diagnostics, field, value) => {
 	return valid;
 };
 
+const validateObservability: ValidatorFn = (
+	diagnostics,
+	field,
+	value,
+	topLevelEnv,
+) => {
+	if (value === undefined) {
+		return true;
+	}
+
+	if (typeof value !== "object") {
+		diagnostics.errors.push(
+			`"${field}" should be an object but got ${JSON.stringify(value)}.`
+		);
+		return false;
+	}
+
+	let isValid = true;
+
+	isValid = validateRequiredProperty(
+		diagnostics,
+		field,
+		"enabled",
+		(value as Observability).enabled,
+		"boolean"
+	) && isValid;
+
+	isValid = validateOptionalProperty(
```

Comment:
A developer could set this as a negative value right? We'd rely on the API to enforce?

Reply:
Yes we'd rely on API to enforce but I'll add client side validation too